channeldb/migration14/migration.go

```diff
@@ -0,0 +1,170 @@
+				return nil
+			}
+
+			chainBucket := nodeChanBucket.NestedReadWriteBucket(
+				chainHash,
+			)
+			if chainBucket == nil {
+				return fmt.Errorf("unable to read "+
+					"bucket for chain=%x", chainHash)
+			}
+
+			return chainBucket.ForEach(func(chanPoint, v []byte) error {
+				// If there's a value, it's not a bucket so
+				// ignore it.
+				if v != nil {
+					return nil
+				}
+
+				// Once we've found a valid channel bucket,
+				// we'll extract it from the node's chain
+				// bucket.
+				chanBucket := chainBucket.NestedReadWriteBucket(
+					chanPoint,
+				)
+				if chanBucket == nil {
+					return fmt.Errorf("not bucket for "+
+						"channel: %x", chanPoint)
+				}
+
+				// At this point, we have the channel bucket
+				// now, so we'll check to see if this channel
+				// has a pending commitment or not. If it
+				// doesn't, then we can go on as there's
+				// nothing to migrate.
+				commitDiffBytes := chanBucket.Get(commitDiffKey)
+				if commitDiffBytes == nil {
+					return nil
+				}
+
+				// Now that we have the commit diff in the
+				// _old_ encoding, we'll write it back to disk
+				// using the new encoding which has a length
+				// prefix in front of the CommitSig.
+				commitDiff, err := deserializeCommitDiffLegacy(
+					bytes.NewReader(commitDiffBytes),
+				)
+				if err != nil {
+					return err
+				}
+
+				var b bytes.Buffer
+				err = serializeCommitDiff(&b, commitDiff)
+				if err != nil {
+					return err
+				}
+
+				err = chanBucket.Put(
+					commitDiffKey, b.Bytes(),
+				)
+				if err != nil {
+					return err
+				}
+
+				// With the commit diff migrated, we'll now
+				// check to see if there're any un-acked
+				// updates we need to migrate as well.
+				updateBytes := chanBucket.Get(
+					unsignedAckedUpdatesKey,
+				)
+				if updateBytes == nil {
+					return nil
+				}
+
+				// We have un-acked updates we need to migrate
+				// so we'll decode then re-encode them here
+				// using the new format.
+				legacyUnackedUpdates, err := deserializeLogUpdatesLegacy(
+					bytes.NewReader(updateBytes),
+				)
+				if err != nil {
+					return err
+				}
+
+				var ub bytes.Buffer
+				err = serializeLogUpdates(
+					&ub, legacyUnackedUpdates,
+				)
+				if err != nil {
+					return err
+				}
+
+				return chanBucket.Put(
+					unsignedAckedUpdatesKey,
+					ub.Bytes(),
+				)
+			})
+		})
+	})
+	if err != nil {
+		return err
+	}
+
+	// Finally, we'll update the pending network results as well.
+	networkResults := tx.ReadWriteBucket(networkResultStoreBucketKey)
+	if networkResults == nil {
+		return nil
+	}
+
+	return networkResults.ForEach(func(k, v []byte) error {
+		oldResult, err := deserializeNetworkResultLegacy(
+			bytes.NewReader(v),
+		)
+		if err != nil {
+			return err
+		}
+
+		var newResultBuf bytes.Buffer
+		err = serializeNetworkResult(
+			&newResultBuf, oldResult,
+		)
+		if err != nil {
+			return err
+		}
+
+		return networkResults.Put(k, newResultBuf.Bytes())
```

Comment:
same comment about not modifying bucket during `ForEach`. This one definitely looks unsafe.